src/mesido/financial_mixin.py

```diff
@@ -95,6 +97,158 @@ def pre(self):
+                    if carrier_name is not None:
+                        self._asset_revenue_map[asset] = [var_name]
+                        self.__asset_revenue_var[var_name] = ca.MX.sym(var_name)
+                        try:
+                            price_profile = self.get_timeseries(
+                                f"{carrier_name}.price_profile"
+                            ).values
+                        except KeyError:
+                            price_profile = np.ones(len(self.times()))
+                        avg_price = np.mean(price_profile) if np.mean(price_profile) else 1.0
+                        self.__asset_revenue_nominals[var_name] = avg_price * nominal_power
+                        self.__asset_revenue_bounds[var_name] = (0.0, np.inf)
+
+            # Find the assets that exchange energy with other owners and create a revenue variable for them as well
+            # We loop over the assets
+            for asset in assets:
+                # This is something extra I created. It is a nested list in the form of:
+                # [[asset1.HeatIn, asset2.HeatOut], [asset3.HeatIn, asset4.HeatOut], ...]
+                conn = self.get_connections()
+                # We loop over the connections (yes this is inefficient)
+                for connection in conn:
+                    # Here we check if our asset is in the connection hence we find a connected asset
+                    if asset == connection[0].split(".")[0] or asset == connection[1].split(".")[0]:
+                        other_asset = (
+                            connection[0].split(".")[0]
+                            if asset == connection[1].split(".")[0]
+                            else connection[1].split(".")[0]
+                        )
+                        # This bit of code is used to go from for example:
+                        # HeatExchanger.Secondary.HeatIn -> Secondary.HeatIn
+                        temp = (
+                            connection[1].split(".")[1:]
+                            if asset == connection[1].split(".")[0]
+                            else connection[0].split(".")[1:]
+                        )
+                        port = temp[0]
+                        for x in temp[1:]:
+                            port = port + "." + x
+
+                        # Here we check that the connected asset is not also owned by the same entity.
+                        # If the other asset belongs to an organization we create a revenue variable
+                        # to represent the cashflow associated with the energy exchanged between the two
+                        # organizations.
+                        if other_asset not in assets:
+                            var_name = f"{asset}__revenue_{port}"
+                            nominal_power = None
+                            carrier_name = None
+                            for _id, attr in self.get_electricity_carriers().items():
```

Comment:
Potentially to generalise for commodities: 
we could create a map commodity_var_map = {"heat": "Heat", "gas": "Q", "electricity": "Power"}
we could loop over the commodities = ["heat", "gas", "electricity"]

then a nested loop for this line could be done like: 
for _id, attr  in getattr(self, f"get_{commodity}_carriers")().items():

with the Power, Q and Heat replaced by commodity_var_map[commodity].

This can also be done for the ccreation of the variables line 119 to 147